diff --git a/sys/dev/rtwn/usb/rtwn_usb_attach.h b/sys/dev/rtwn/usb/rtwn_usb_attach.h
index f5d9e97b66e..2a7c962838a 100644
--- a/sys/dev/rtwn/usb/rtwn_usb_attach.h
+++ b/sys/dev/rtwn/usb/rtwn_usb_attach.h
@@ -145,6 +145,7 @@ static const STRUCT_USB_HOST_ID rtwn_devs[] = {
 	RTWN_RTL8812AU_DEV(REALTEK,		RTL8812AU_2),
 	RTWN_RTL8812AU_DEV(SENAO,		EUB1200AC),
 	RTWN_RTL8812AU_DEV(SITECOMEU,		WLA7100),
+	RTWN_RTL8812AU_DEV(TENDA2,		U12),
 	RTWN_RTL8812AU_DEV(TPLINK,		T4U),
 	RTWN_RTL8812AU_DEV(TPLINK,		T4UV2),
 	RTWN_RTL8812AU_DEV(TPLINK,		T4UHV1),
diff --git a/sys/dev/usb/usbdevs b/sys/dev/usb/usbdevs
index 9c2206c6e74..2153720b7b5 100644
--- a/sys/dev/usb/usbdevs
+++ b/sys/dev/usb/usbdevs
@@ -777,6 +777,7 @@ vendor ARDUINO		0x2341	Arduino SA
 vendor TPLINK		0x2357	TP-Link
 vendor WESTMOUNTAIN	0x2405	West Mountain Radio
 vendor TRIPPLITE	0x2478	Tripp-Lite
+vendor TENDA2		0x2604	Tenda
 vendor HIROSE		0x2631	Hirose Electric
 vendor NHJ		0x2770	NHJ
 vendor THINGM		0x27b8	ThingM
@@ -4667,6 +4668,9 @@ product TELIT UC864G		0x1004	UC864G 3G modem
 /* Ten X Technology, Inc. */
 product TENX UAUDIO0		0xf211	USB audio headset
 
+/* Tenda products */
+product TENDA2 U12		0x0012	Tenda U12
+
 /* ThingM products */
 product THINGM BLINK1		0x01ed	USB notification light
 
